### PR TITLE
[Lint] Use flake8 instead of pep8

### PR DESCRIPTION
**Pull request by lukeyeager**
`flake8` checks for logic errors as well as stylistic issues. It also uses `pycodestyle` internally instead of the outdated `pep8` (thanks to @colesbury for pointing this out in https://github.com/pytorch/pytorch/pull/646).

While switching over, I fixed a few easy rules. Here are the errors `flake8` reported before the fixes:
```
428   F401 'torchvision' imported but unused
311   F405 'InplaceFunction' may be undefined, or defined from star imports: torch.autograd._functions
50    E402 module level import not at top of file
48    F821 undefined name 'error'
34    F841 local variable 'e' is assigned to but never used
26    E305 expected 2 blank lines after class or function definition, found 1
23    F403 'from torch.autograd._functions import *' used; unable to detect undefined names
6     F811 redefinition of unused 'test_serialization' from line 385
3     E731 do not assign a lambda expression, use a def
2     E721 do not compare types, use 'isinstance()'
1     F812 list comprehension redefines 'name' from line 50
```
And here is the trimmed down list:
```
428   F401 'torchvision' imported but unused
308   F405 'InplaceFunction' may be undefined, or defined from star imports: torch.autograd._functions
50    E402 module level import not at top of file
48    F821 undefined name 'error'
34    F841 local variable 'e' is assigned to but never used
26    E305 expected 2 blank lines after class or function definition, found 1
23    F403 'from torch.autograd._functions import *' used; unable to detect undefined names
2     E721 do not compare types, use 'isinstance()'
```
^ Those rules are the ones I've placed in `tox.ini`.

If you like `flake8`, I'll take a look at adding it to `pytorch/vision` as well (https://github.com/pytorch/vision/pull/65#issuecomment-282256807).

### CODE REVIEW

File: test/test_cuda.py [hunk 1]
```diff
@@ -527,16 +527,6 @@ def test_cat_autogpu(self):
         z = torch.cat([x, y], 0)
         self.assertEqual(z.get_device(), x.get_device())
 
-    def test_serialization(self):
```
Comment (apaszke) [inline test/test_cuda.py]: Why did you remove that test?

**Review by apaszke — CHANGES_REQUESTED** [review]
You removed some tests and Variable methods. Can you please add them back?

File: test/test_cuda.py (hunk 1 above)
Comment (lukeyeager) [inline test/test_cuda.py]: There are two tests with the same name. I removed the older one.

Comment (lukeyeager) [inline test/test_cuda.py]: Happy to rename it instead, but if you look at the two definitions it seems like the first one is what you want. Unfortunately, only the second test is running currently, since it clobbers the first definition.

Comment (apaszke) [inline test/test_cuda.py]: Oh, so the Variable methods were duplicates too?

Yes, could you rename the test? Thank you!

Comment (lukeyeager) [inline test/test_cuda.py]: Resolved by renaming the first test instead.

File: test/test_torch.py [hunk 2]
```diff
@@ -1653,14 +1655,6 @@ def reference(x, k, o3, o32):
         self._test_conv_corr_eq(lambda x, k: torch.xcorr3(x, k), reference)
 
     @unittest.skip("Not implemented yet")
-    def test_xcorr3_xcorr2_eq(self):
```
Comment (fmassa) [inline test/test_torch.py]: This test is not the same as [this one, which has the same name](https://github.com/lukeyeager/pytorch/blob/591a25490732088c9bb5bc11835af041bc58a9e9/test/test_torch.py#L1650-L1655). I think we should change the name instead of removing it.

Comment (lukeyeager) [inline test/test_torch.py]: Name suggestions? I didn't think it mattered much since they're both perma-skipped anyway.

Comment (lukeyeager) [inline test/test_torch.py]: Where can I read up on `xcorr2()`? I don't see it defined anywhere?

Comment (soumith) [inline test/test_torch.py]: it's the torch cross-correlation function not implemented in pytorch yet

Comment (lukeyeager) [inline test/test_torch.py]: Oh I see. The function to be tested doesn't even exist yet. What should the two different functions be called?

Comment (fmassa) [inline test/test_torch.py]: the one with the 'F' seems to be the full correlation, and the other `valid`. Maybe add a `_full` somewhere? :)

Comment (lukeyeager) [inline test/test_torch.py]: Found it. "full" vs "valid" - https://github.com/torch/torch7/blob/cd8de6c67d/lib/TH/generic/THTensorConv.c#L499-L511

Comment (lukeyeager) [inline test/test_torch.py]: #resolved

**Review by soumith — APPROVED** [review]

**Comment by soumith** [comment]
Thanks Luke!
